extensions/wikia/Wall/VoteHelper.class.php

```diff
@@ -1,45 +1,40 @@
 <?php
 
-// TODO: move this to services
-
+/**
+ * Class VoteHelper
+ */
 class VoteHelper {
 	protected $userId;
 	protected $pageId;
 	protected $userIP = '';
 
-	function __construct( $user = null, $pageId = 0 ) {
+	function __construct( User $user = null, $pageId = 0 ) {
 		global $wgRequest;
-		wfProfileIn( __METHOD__ );
-		$this->userIP = $wgRequest->getIP();
-		if ( $user->isLoggedIn() ) {
-			$this->userId = $user->getId();
-		} else {
-			$this->userId = 0;
-		}
```

Comment:
A user who is not logged in will have an ID of zero (the ```isLoggedIn``` method just checks if ```$this->id``` != 0) so we don't need this conditional.